packages/ui/src/hooks/use-social.ts

```diff
@@ -3,18 +3,20 @@ import { useCallback, useContext } from 'react';
 import { invokeSocialSignIn } from '@/apis/social';
 
 import useApi from './use-api';
+import useAppleAuth, { isAppleConnector } from './use-apple-auth';
 import { PageContext } from './use-page-context';
 import useTerms from './use-terms';
 import { getLogtoNativeSdk, isNativeWebview, generateState, storeState } from './utils';
 
 const useSocial = () => {
   const { experienceSettings } = useContext(PageContext);
   const { termsValidation } = useTerms();
+  const appleAuth = useAppleAuth();
 
   const { run: asyncInvokeSocialSignIn } = useApi(invokeSocialSignIn);
 
   const invokeSocialSignInHandler = useCallback(
-    async (connectorId: string, callback?: () => void) => {
+    async (connectorId: string, target: string, callback?: () => void) => {
```

Comment:
Since we have all connectors loaded in the context, one alteration is query out the connector target using the connectorId, so no need to pass in the target any more. Not sure which one is better.

Reply:
actually i would like to pass the whole connector object as the param since it's just an address

Reply:
LOG-2725